lib/matplotlib/figure.py

```diff
@@ -3119,6 +3121,9 @@ def draw(self, renderer):
 
         artists = self._get_draw_artists(renderer)
         try:
+            """For every new Figure, seed_increment of gc is reset to 0"""
+            GraphicsContextBase().reset_seed_increment()
```

Comment:
Ah yes that would make the code much cleaner. I thought that having the seed in a different class would not be a good idea, have made that change now. 

And also yes the issue started after making seed a seperate global variable. I have found the cause of it to be :
https://github.com/AryanSheka/matplotlib/blob/d0b0c16ae33acd6aef79e7bc7356253112d3f6c6/lib/matplotlib/artist.py#L211
Since path.sketch was a 4-tuple before we made the seed a seperate global variable and hence chaging it to a 4-tuple in the following manner fixes the issue :
https://github.com/AryanSheka/matplotlib/blob/44816416b6741aefdd27aaa8c2c9d44a3d463ece/lib/matplotlib/artist.py#L211